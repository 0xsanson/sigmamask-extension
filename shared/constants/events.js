export const CONTROLLER_CONNECTION_EVENTS = {
  UPDATE: 'update',
  SEND_UPDATE: 'sendUpdate',
  ACTION: 'action',
  SEND_ACTION: 'sendAction',
  HARDWARE_CALL: 'hardwareCall',
  SEND_HARDWARE_CALL: 'sendHardwareCall',
};
