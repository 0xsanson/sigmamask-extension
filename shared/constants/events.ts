export enum ControllerConnectionEvents {
  update = 'update',
  sendUpdate = 'sendUpdate',
  action = 'action',
  sendAction = 'sendAction',
  hardwareCall = 'hardwareCall',
  sendHardwareCall = 'sendHardwareCall',
}
