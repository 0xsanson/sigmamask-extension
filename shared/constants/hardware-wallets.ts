/**
 * Hardware wallets supported by MetaMask.
 */
export enum HardwareKeyringType {
  ledger = 'Ledger Hardware',
  trezor = 'Trezor Hardware',
  lattice = 'Lattice Hardware',
  qr = 'QR Hardware Wallet Device',
}

export enum HardwareKeyringNames {
  ledger = 'Ledger',
  trezor = 'Trezor',
  lattice = 'Lattice1',
  qr = 'QR',
}

export enum HardwareDeviceNames {
  ledger = 'ledger',
  trezor = 'trezor',
  lattice = 'lattice',
  qr = 'QR Hardware',
}

export enum HardwareTransportStates {
  none = 'NONE',
  verified = 'VERIFIED',
  deviceOpenFailure = 'DEVICE_OPEN_FAILURE',
  unknownFailure = 'UNKNOWN_FAILURE',
}

export enum HardwareAffiliateLinks {
  ledger = 'https://shop.ledger.com/?r=17c4991a03fa',
  gridplus = 'https://gridplus.io/?afmc=7p',
  trezor = 'https://shop.trezor.io/product/trezor-one-black?offer_id=35&aff_id=11009',
  keystone = 'https://shop.keyst.one/?rfsn=6088257.656b3e9&utm_source=refersion&utm_medium=affiliate&utm_campaign=6088257.656b3e9',
  airgap = 'https://airgap.it/',
  coolwallet = 'https://www.coolwallet.io/',
  dcent = 'https://dcentwallet.com/',
}

export enum HardwareAffiliateTutorialLinks {
  ledger = 'https://support.ledger.com/hc/en-us/articles/4404366864657-Set-up-and-use-MetaMask-to-access-your-Ledger-Ethereum-ETH-account?docs=true',
  gridplus = 'https://docs.gridplus.io/setup/metamask',
  trezor = 'https://wiki.trezor.io/Apps:MetaMask',
  keystone = 'https://support.keyst.one/3rd-party-wallets/eth-and-web3-wallets-keystone/bind-metamask-with-keystone',
  airgap = 'https://support.airgap.it/guides/metamask/',
  coolwallet = 'https://www.coolwallet.io/metamask-step-by-step-guides/',
  dcent = 'https://medium.com/dcentwallet/dcent-wallet-now-supports-qr-based-protocol-to-link-with-metamask-57555f02603f',
}

/**
 * Used for setting the users preference for ledger transport type
 */
export enum LedgerTransportTypes {
  live = 'ledgerLive',
  webhid = 'webhid',
  u2f = 'u2f',
}

export enum WebHIDConnectedStatuses {
  connected = 'connected',
  notConnected = 'notConnected',
  unknown = 'unknown',
}

export const LEDGER_USB_VENDOR_ID = '0x2c97';

export const U2F_ERROR = 'U2F';

export const LEDGER_LIVE_PATH = `m/44'/60'/0'/0/0`;
export const MEW_PATH = `m/44'/60'/0'`;
export const BIP44_PATH = `m/44'/60'/0'/0`;
export const LEDGER_HD_PATHS = [
  { name: 'Ledger Live', value: LEDGER_LIVE_PATH },
  { name: 'Legacy (MEW / MyCrypto)', value: MEW_PATH },
  { name: `BIP44 Standard (e.g. MetaMask, Trezor)`, value: BIP44_PATH },
];

export const LATTICE_STANDARD_BIP44_PATH = `m/44'/60'/0'/0/x`;
export const LATTICE_LEDGER_LIVE_PATH = `m/44'/60'/x'/0/0`;
export const LATTICE_MEW_PATH = `m/44'/60'/0'/x`;
export const LATTICE_HD_PATHS = [
  {
    name: `Standard (${LATTICE_STANDARD_BIP44_PATH})`,
    value: LATTICE_STANDARD_BIP44_PATH,
  },
  {
    name: `Ledger Live (${LATTICE_LEDGER_LIVE_PATH})`,
    value: LATTICE_LEDGER_LIVE_PATH,
  },
  { name: `Ledger Legacy (${LATTICE_MEW_PATH})`, value: LATTICE_MEW_PATH },
];

export const TREZOR_TESTNET_PATH = `m/44'/1'/0'/0`;
export const TREZOR_HD_PATHS = [
  { name: `BIP44 Standard (e.g. MetaMask, Trezor)`, value: BIP44_PATH },
  { name: `Trezor Testnets`, value: TREZOR_TESTNET_PATH },
];

export const HD_PATHS = {
  ledger: LEDGER_HD_PATHS,
  lattice: LATTICE_HD_PATHS,
  trezor: TREZOR_HD_PATHS,
};

export const DEFAULT_HD_PATHS = {
  trezor: BIP44_PATH,
  ledger: LEDGER_LIVE_PATH,
  lattice: BIP44_PATH,
};
