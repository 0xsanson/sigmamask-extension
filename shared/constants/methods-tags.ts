/**
 * This is a list of methods that require the globally selected network
 * to match the dapp selected network before being processed. These can
 * be for UI/UX reasons where the currently selected network is displayed
 * in the confirmation even though it will be submitted on the correct
 * network for the dapp. It could also be that a method expects the
 * globally selected network to match some value in the request params itself.
 */
export const methodsRequiringNetworkSwitch = Object.freeze([
  'eth_sendTransaction' as const,
  'eth_sendRawTransaction' as const,
  'wallet_switchEthereumChain' as const,
  'wallet_addEthereumChain' as const,
  'wallet_watchAsset' as const,
  'eth_signTypedData_v4' as const,
  'personal_sign' as const,
]);

/**
 * This is a list of methods that can cause a confirmation to be
 * presented to the user. Note that some of these methods may
 * only sometimes cause a confirmation to appear.
 */
export const methodsWithConfirmation = Object.freeze([
  ...methodsRequiringNetworkSwitch,
  'wallet_requestPermissions' as const,
  'wallet_requestSnaps' as const,
  'wallet_snap' as const,
  'wallet_invokeSnap' as const,
  'eth_decrypt' as const,
  'eth_sign' as const,
  'eth_requestAccounts' as const,
  'eth_getEncryptionPublicKey' as const,
]);
