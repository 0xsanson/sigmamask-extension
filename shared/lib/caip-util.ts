export function isCaipChainId(caipChainId: string): boolean {
  return caipChainId.startsWith('eip155:');
}
export function getCaipChainIdFromEthChainId(ethChainId: string): string {
  const chainIdDecimal = ethChainId.startsWith('0x')
    ? parseInt(ethChainId, 16).toString(10)
    : ethChainId;
  return `eip155:${chainIdDecimal}`;
}

export function getEthChainIdDecFromCaipChainId(caipChainId: string): string {
  return caipChainId.replace(/^eip155:/, '')
}

export function getEthChainIdHexFromCaipChainId(caipChainId: string): string {
  const chainIdDecimal = getEthChainIdDecFromCaipChainId(caipChainId)
  return `0x${parseInt(chainIdDecimal, 10).toString(16)}`;
}

export function getEthChainIdNumFromCaipChainId(caipChainId: string): number {
  const chainIdDecimal = getEthChainIdDecFromCaipChainId(caipChainId)
  return parseInt(chainIdDecimal, 10)
}
