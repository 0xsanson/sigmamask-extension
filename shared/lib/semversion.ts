/**
 * A simple utility to parse and compare semver versions.
 */

import type ExtensionPlatform from '../../app/scripts/platforms/extension';

/**
 * A semver version object.
 */
export type SemVersion = {
  major: number;
  minor: number;
  patch: number;
};

/**
 * Parse a semver version string into a SemVersion object.
 *
 * @param version - The version string to parse, of the form "major.minor.patch". e.g. "1.2.3"
 * @returns
 */
export const parseVersion = (version: string = ''): SemVersion | null => {
  const [major, minor, patch] = version.split('.').map(Number);

  // if version is not a valid semver, return false
  if ([major, minor, patch].some((num) => isNaN(num) || num < 0)) {
    return null;
  }
  return { major, minor, patch };
};

/**
 * Compare two SemVersion objects.
 *
 * @param a
 * @param b
 * @returns negative if a < b, zero if a === b, positive if a > b
 */
export const compareVersions = (a: SemVersion, b: SemVersion): number => {
  if (a.major !== b.major) {
    return a.major - b.major;
  }
  if (a.minor !== b.minor) {
    return a.minor - b.minor;
  }
  return a.patch - b.patch;
};

/**
 * Get the version of the extension.
 */
export const getMetamaskVersion = () =>
  parseVersion((global.platform as ExtensionPlatform).getVersion());
