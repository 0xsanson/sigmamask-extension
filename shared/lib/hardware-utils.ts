import { isManifestV3 } from '../modules/mv3.utils';
import { LedgerTransportTypes } from '../constants/hardware-wallets';

export const getDefaultLedgerTransportType = (): LedgerTransportTypes => {
  if (isManifestV3) {
    // Then enforce the use of WebHID as the default, despite
    // the current context not supporting window.navigator.hid
    return LedgerTransportTypes.webhid;
  }

  return Object.hasOwnProperty.call(window.navigator, 'hid')
    ? LedgerTransportTypes.webhid
    : LedgerTransportTypes.u2f;
};

export const getRecommendedLedgerTransportType = (): LedgerTransportTypes => {
  // Currently the recommended ledger transport type is the default
  return getDefaultLedgerTransportType();
};

type LedgerTranslations = Record<LedgerTransportTypes, string>;
type LedgerTransportOption = {
  name: LedgerTranslations[LedgerTransportTypes];
  value: LedgerTransportTypes;
};

export const getLedgerTransportOption = (
  transportType: LedgerTransportTypes,
  translations: LedgerTranslations,
): LedgerTransportOption => {
  return {
    name: translations[transportType],
    value: transportType,
  };
};
