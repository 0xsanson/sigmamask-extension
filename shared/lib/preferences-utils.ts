import { isManifestV3 } from '../modules/mv3.utils';
import { LedgerTransportTypes } from '../constants/hardware-wallets';

export const getInitLedgerTransportType = (): string => {
  if (isManifestV3) {
    // Then enforce the use of WebHID as the default, despite
    // the current context not supporting window.navigator.hid
    return LedgerTransportTypes.webhid;
  }

  return Object.hasOwnProperty.call(window.navigator, 'hid')
    ? LedgerTransportTypes.webhid
    : LedgerTransportTypes.u2f;
};
