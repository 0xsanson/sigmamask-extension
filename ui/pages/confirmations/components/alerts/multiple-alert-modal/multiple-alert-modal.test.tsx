import React from 'react';
import configureMockStore from 'redux-mock-store';
import { Severity } from '../../../../../helpers/constants/design-system';
import { renderWithProvider } from '../../../../../../test/lib/render-helpers';
import {
  MultipleAlertModal,
  MultipleAlertModalProps,
} from './multiple-alert-modal';

describe('MultipleAlertModal', () => {
  const ownerIdMock = '123';
  const handleButtonClickMock = jest.fn();
  const onCloseMock = jest.fn();
  const fromAlertKeyMock = 'from';
  const alertsMock = [
    {
      key: fromAlertKeyMock,
      severity: Severity.Warning,
      message: 'Alert 1',
      reason: 'Reason 1',
      alertDetails: ['Detail 1', 'Detail 2'],
    },
    { key: 'data', severity: Severity.Danger, message: 'Alert 2' },
  ];

  const mockState = {
    confirmAlerts: {
      alerts: { [ownerIdMock]: alertsMock },
      confirmed: { [ownerIdMock]: { [fromAlertKeyMock]: false, data: false } },
    },
  };
  const mockStore = configureMockStore([])(mockState);

  const defaultProps: MultipleAlertModalProps = {
    ownerId: ownerIdMock,
    handleButtonClick: handleButtonClickMock,
    alertKey: fromAlertKeyMock,
    onClose: onCloseMock,
  };

  it('renders the multiple alert modal', () => {
    const { container } = renderWithProvider(
      <MultipleAlertModal {...defaultProps} />,
      mockStore,
    );

    expect(container).toMatchSnapshot();
  });
});
