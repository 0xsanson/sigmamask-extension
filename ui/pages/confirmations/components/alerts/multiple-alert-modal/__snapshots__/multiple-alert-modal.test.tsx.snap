// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultipleAlertModal renders the multiple alert modal 1`] = `<div />`;
